### PR TITLE
 Rewrote model, added tests, card object id - card id mapping

### PR DESCRIPTION
**Pull request by mrprizmo**

### CODE REVIEW

File: include/model_card_manager.hpp [hunk 1]
```diff
@@ -1,15 +1,19 @@
...
 #include "model_card.hpp"
 
 namespace meow {
 
+std::unique_ptr<model::MonsterCard> card_to_monster(std::unique_ptr<model::Card> card);
+
 struct CardManager {
 private:
     std::vector<std::unique_ptr<model::CardInfo>> cards_instances;
+    mutable std::map<std::size_t, std::size_t> obj_id_to_card_id;
```
Comment (mmalofeev) [inline include/model_card_manager.hpp]: why map and not unordered_map?

File: include/shared_game_state.hpp [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    std::size_t cur_turn = std::numeric_limits<std::size_t>::max();
+
+public:
+    void set_player_list(const std::vector<std::size_t> &users);
+
+    std::size_t get_current_user_id() const;
+
+    std::size_t get_number_of_players() const;
+
+    std::size_t get_card_id_from_deck();
```
Comment (mmalofeev) [inline include/shared_game_state.hpp]: const

File: include/shared_game_state.hpp [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+    Player *get_player_by_player_id(std::size_t player_id);
+
+    Player *get_player_by_user_id(std::size_t user_id);
+
+    std::size_t get_player_position_by_user_id(std::size_t user_id) const;
```
Comment (mmalofeev) [inline include/shared_game_state.hpp]: const

File: src/game_state.cpp [hunk 4]
```diff
@@ -0,0 +1,298 @@
...
+    }
+
+    return std::unique_ptr<ManagementState>(this);
+}
+
+std::unique_ptr<GameState>
+ManagementState::throw_card(std::size_t user_id, std::size_t card_obj_id) {
+    Player *player = shared_state->get_player_by_user_id(user_id);
+
+    if ([[maybe_unused]] auto card = player->drop_card_from_hand_by_id(card_obj_id)) {
```
Comment (mmalofeev) [inline src/game_state.cpp]: `if (player->drop_card_from_hand_by_id(card_obj_id))`

File: include/shared_game_state.hpp (hunk 3 above)
Comment (mrprizmo) [reply include/shared_game_state.hpp]: not marked as const because game_state must have access to change player fields

**Pull request merged**